node/src/components/event_stream_server/tests.rs

```diff
@@ -0,0 +1,899 @@
+    /// Returns a new `ServerSyncBehavior` with a 1 millisecond delay per event, and no clients.
+    fn new() -> Self {
+        ServerSyncBehavior {
+            has_delay_between_events: true,
+            clients: Vec::new(),
+        }
+    }
+
+    /// Removes the delay between events.
+    fn with_no_delay_between_events(mut self) -> Self {
+        self.has_delay_between_events = false;
+        self
+    }
+
+    /// Adds a client sync behavior, specified for the client to connect to the server just before
+    /// `id` is emitted.
+    fn add_client_sync_before_event(&mut self, id: Id) -> Arc<Barrier> {
+        let (client_behavior, barrier) = ClientSyncBehavior::new(id);
+        self.clients.push(client_behavior);
+        barrier
+    }
+
+    /// Waits for all clients which specified they wanted to join just before the given event ID.
+    async fn wait_for_clients(&self, id: Id) {
+        for client_behavior in &self.clients {
+            if client_behavior.join_before_event == id {
+                debug!("server waiting before event {}", id);
+                client_behavior.barrier.wait().await;
+                debug!("server waiting for client to connect before event {}", id);
+                client_behavior.barrier.wait().await;
+                debug!("server finished waiting before event {}", id);
+            }
+        }
+    }
+
+    /// Sleeps if `self` was set to enable delays between events.
+    async fn sleep_if_required(&self) {
+        if self.has_delay_between_events {
+            time::sleep(DELAY_BETWEEN_EVENTS).await;
+        }
+    }
+}
+
+/// A helper to allow the server to be kept alive until a specific call to stop it.
+#[derive(Clone)]
+struct ServerStopper {
+    should_stop: Arc<AtomicBool>,
+    notifier: Arc<Notify>,
+}
+
+impl ServerStopper {
+    fn new() -> Self {
+        ServerStopper {
+            should_stop: Arc::new(AtomicBool::new(false)),
+            notifier: Arc::new(Notify::new()),
+        }
+    }
+
+    /// Returns whether the server should stop now or not.
+    fn should_stop(&self) -> bool {
+        self.should_stop.load(Ordering::SeqCst)
+    }
+
+    /// Waits until the server should stop.
+    async fn wait(&self) {
+        while !self.should_stop() {
+            self.notifier.notified().await;
+        }
+    }
+
+    /// Tells the server to stop.
+    fn stop(&self) {
+        self.should_stop.store(true, Ordering::SeqCst);
+        self.notifier.notify_one();
+    }
+}
+
+impl Drop for ServerStopper {
+    fn drop(&mut self) {
+        self.stop();
+    }
+}
+
+struct TestFixture {
+    storage_dir: TempDir,
+    protocol_version: ProtocolVersion,
+    events: Vec<SseData>,
+    first_event_id: Id,
+    server_join_handle: Option<JoinHandle<()>>,
+    server_stopper: ServerStopper,
+}
+
+impl TestFixture {
+    /// Constructs a new `TestFixture` including `EVENT_COUNT` random events ready to be served.
+    fn new(rng: &mut TestRng) -> Self {
```

Comment:
Maybe pass in `EVENT_COUNT` here, since we might want to use different settings for different tests?

Reply:
None of the tests need this flexibility, so I'd rather avoid adding an extra parameter.  We can always do it later if we add new tests which might need that.